src/bokeh/core/property/data_class.py

```diff
@@ -0,0 +1,68 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) Anaconda, Inc., and Bokeh Contributors.
+# All rights reserved.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+""" Provide the dataclass property.
+"""
+
+#-----------------------------------------------------------------------------
+# Boilerplate
+#-----------------------------------------------------------------------------
+from __future__ import annotations
+
+import logging # isort:skip
+log = logging.getLogger(__name__)
+
+#-----------------------------------------------------------------------------
+# Imports
+#-----------------------------------------------------------------------------
+
+# Standard library imports
+from dataclasses import is_dataclass
+from typing import Any
+
+# Bokeh imports
+from .bases import Property
+
+#-----------------------------------------------------------------------------
+# Globals and constants
+#-----------------------------------------------------------------------------
+
+__all__ = (
+    'Dataclass',
+)
+
+
+#-----------------------------------------------------------------------------
+# General API
+#-----------------------------------------------------------------------------
+
+class Dataclass(Property["dataclass"]):
```

Comment:
This should be either:
```suggestion
class Dataclass(Property[object]):
```
or perhaps even preferably:
```suggestion
T = TypeVar("T", bound=object)
class Dataclass(Property[T]):
```
There's also `Object[T]` property type, that `Instance[T]` inherits from, that could be used a as base for specific object related property types.

Reply:
Done in  a85047771